### PR TITLE
feat(module-1): Implementing ride-cleansing and rides-and-fares

### PR DESCRIPTION
**Pull request by gerson23**
* Implemented ride-cleaning through filtering in only rides that started and finished within NYC range
* In rides-and-fares, TaxiRide and TaxiFare records may come in different order, so we need to have state for each
  * Whenever a pair is paired, clean the state that was set

### CODE REVIEW

File: rides-and-fares/src/main/java/org/apache/flink/training/exercises/ridesandfares/RidesAndFaresExercise.java [hunk 1]
```diff
@@ -99,19 +101,39 @@ public static void main(String[] args) throws Exception {
     public static class EnrichmentFunction
             extends RichCoFlatMapFunction<TaxiRide, TaxiFare, RideAndFare> {
 
+        private ValueState<TaxiRide> startState;
```
Comment (ness-PavelKhokhlov) [inline rides-and-fares/src/main/java/org/apache/flink/training/exercises/ridesandfares/RidesAndFaresExercise.java]: should be `transient`
`private transient`
pls see details:
https://stackoverflow.com/a/72558397

File: rides-and-fares/src/main/java/org/apache/flink/training/exercises/ridesandfares/RidesAndFaresExercise.java [hunk 2]
```diff
@@ -99,19 +101,39 @@ public static void main(String[] args) throws Exception {
...
+        private ValueState<TaxiFare> fareState;
```
Comment (ness-PavelKhokhlov) [inline rides-and-fares/src/main/java/org/apache/flink/training/exercises/ridesandfares/RidesAndFaresExercise.java]: should be `transient`
`private transient`
pls see details:
https://stackoverflow.com/a/72558397

**Review by ness-PavelKhokhlov — CHANGES_REQUESTED** [review]
Hello @gerson23 !
Please see my comments.

File: rides-and-fares/src/main/java/org/apache/flink/training/exercises/ridesandfares/RidesAndFaresExercise.java (hunk 1 above)
Comment (gerson23) [reply rides-and-fares/src/main/java/org/apache/flink/training/exercises/ridesandfares/RidesAndFaresExercise.java]: Thanks! Added transient to both states.

**Review by ness-PavelKhokhlov — APPROVED** [review]

**Pull request merged**